packages/fuel-indexer/src/executor.rs

```diff
@@ -79,75 +83,105 @@ pub fn run_executor<T: 'static + Executor + Send + Sync>(
 ) -> impl Future<Output = ()> {
     // TODO: https://github.com/FuelLabs/fuel-indexer/issues/286
 
-    let start_block = manifest
-        .start_block()
-        .expect("Failed to detect start_block.");
     let end_block = manifest.end_block();
-    if end_block.is_none() {
-        warn!("No end_block specified in manifest. Indexer will run forever.");
-    }
     let stop_idle_indexers = config.stop_idle_indexers;
     let indexer_uid = manifest.uid();
-
-    let fuel_node_addr = if config.indexer_net_config {
-        manifest
-            .fuel_client()
-            .map(|x| x.to_string())
-            .unwrap_or(config.fuel_node.to_string())
-    } else {
-        config.fuel_node.to_string()
-    };
-
     let node_block_page_size = config.node_block_page_size;
 
-    let mut next_cursor = if start_block > 1 {
-        let decremented = start_block - 1;
-        Some(decremented.to_string())
-    } else {
-        None
-    };
+    let fuel_node_addr = manifest
+        .fuel_client()
+        .map(|x| x.to_string())
+        .unwrap_or(config.fuel_node.to_string());
+
+    // Where should we initially start when fetching blocks from the client?
+    let mut cursor = manifest.start_block().map(|x| {
+        if x > 1 {
+            let decremented = x - 1;
+            decremented.to_string()
+        } else {
+            "0".to_string()
+        }
+    });
 
     info!("Indexer({indexer_uid}) subscribing to Fuel node at {fuel_node_addr}");
 
     let client = FuelClient::from_str(&fuel_node_addr).unwrap_or_else(|e| {
         panic!("Indexer({indexer_uid}) client node connection failed: {e}.")
     });
 
+    if end_block.is_none() {
+        warn!("No end_block specified in manifest. Indexer will run forever.");
+    }
+
     async move {
+        // If we reach an issue that continues to fail, we'll retry a few times before giving up, as
+        // we don't want to quit on the first error. But also don't want to waste CPU.
+        //
+        // Note that this count considers _consecutive_ failed calls.
         let mut retry_count = 0;
 
         // If we're testing or running on CI, we don't want indexers to run forever. But in production
-        // let the index operators decide if they want to stop idle indexers. Maybe we can eventually
-        // make this MAX_EMPTY_BLOCK_REQUESTS value configurable
+        // let the indexer service operator decide if they want to stop idle indexers.
+        //
+        // Maybe we can eventually make this MAX_EMPTY_BLOCK_REQUESTS value configurable
+        //
+        // Also note that this count considers _consecutive_ empty block requests.
```

Comment:
Perhaps we can rename `retry_count` and `MAX_EMPTY_BLOCK_REQUESTS` to `consecutive_retries` and `MAX_CONSECUTIVE_EMPTY_BLOCK_RESPONSES`, respectively? It'll make it more self-documenting and it also communicates that we're logging empty block _responses_ to our requests.